Consertar erro ao inicializar banco local

Ao mudar a rota o seguinte erro é retornado:

 ERROR  Erro ao inicializar o banco de dados: [Error: Call to function 'NativeDatabase.execAsync' has been rejected.
→ Caused by: java.lang.NullPointerException: java.lang.NullPointerException] 

Call Stack
  construct (<native>)
  apply (<native>)
  _construct (node_modules\@babel\runtime\helpers\construct.js)
  Wrapper (node_modules\@babel\runtime\helpers\wrapNativeSuper.js)
  construct (<native>)
  _callSuper (node_modules\@babel\runtime\helpers\callSuper.js)
  constructor (node_modules\expo-modules-core\src\errors\CodedError.ts)

--- a/app/(app)/index.tsx
+++ b/app/(app)/index.tsx
@@ -1,29 +1,7 @@
-import { initDB } from "@/database/database";
 import { router } from "expo-router";
-import { useEffect, useState } from "react";
 import { Pressable, StyleSheet, Text, View } from "react-native";
 
 export default function HomePage() {
-  const [isDbReady, setIsDbReady] = useState(false);
-
-  useEffect(() => {
-    const initializeDB = async () => {
-      await initDB();
-      setIsDbReady(true);
-    };
-
-    initializeDB();
-  }, []);
-
-  if (!isDbReady) {
-    return (
-      <View style={styles.container}>
-        <Text style={styles.title}>Cognisus Mobile</Text>
-        <Text style={styles.subtitle}>Loading...</Text>
-      </View>
-    );
-  }
-
   return (
     <View style={styles.container}>
       <Text style={styles.title}>Cognisus Mobile</Text>

--- a/app/(auth)/cadastro.tsx
+++ b/app/(auth)/cadastro.tsx
@@ -64,19 +64,46 @@ export default function Cadastro() {
       if (data.user) {
         const userId = data.user.id;
 
+        // Validação de userId
+        if (!userId) {
+          setLoading(false);
+          return Alert.alert(
+            "Erro",
+            "Não foi possível obter o ID do usuário do Supabase",
+          );
+        }
+
         try {
           const db = await getDB();
+          const createdAt = new Date().toISOString();
+
           await db.runAsync(
             `INSERT INTO profissional (user_id, nome_completo, cpf, email, sync_status, created_at) 
-             VALUES (?, ?, ?, ?, 'synced', datetime('now'))`,
-            [userId, name, cpf, email],
+             VALUES (?, ?, ?, ?, ?, ?)`,
+            [userId, name, cpf, email, "synced", createdAt],
           );
 
           Alert.alert("Sucesso", "Cadastro realizado com sucesso!");
           router.replace("/(app)");
         } catch (dbError) {
-          console.error("Erro ao salvar no banco local", dbError);
-          router.replace("/(app)");
+          console.error("Erro ao salvar no banco local:", dbError);
+
+          // Melhor tratamento de erro
+          if (dbError instanceof Error) {
+            const errorMessage =
+              dbError.message ||
+              "Erro desconhecido ao salvar no banco de dados";
+            console.error("Detalhes do erro:", errorMessage);
+
+            // Verifica se é erro de constraint (CPF ou Email duplicados)
+            if (errorMessage.includes("UNIQUE constraint failed")) {
+              setLoading(false);
+              return Alert.alert(
+                "Erro",
+                "CPF ou Email já cadastrado no sistema",
+              );
+            }
+          }
         }
       }
     } catch (error) {

